pkg/kubelet/kubelet.go

```diff
@@ -711,6 +712,8 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 			runtimeService,
 			imageService,
 			stats.NewLogMetricsService())
+
+		prometheus.MustRegister(klet.StatsProvider.Collector())
```

Comment:
typically, we have put registration of collectors under kubelet/metrics/metrics.go, see the Register function there.

Can you move this under `initializeModules` where it `calls metrics.Register` and pass in the additional collector?